src/core/font_substitutions.js

```diff
@@ -333,38 +287,96 @@ const substitutionMap = new Map([
+        return "Italic";
+      }
+  }
+  return "";
+}
+
 /**
- * Create the src path to use to load a font (see FontFace).
- * @param {Array<String>} prepend A list of font names to search first.
- * @param {String} appendToPrepended A String to append to the list of fonts in
- *  prepend.
- * @param {Array<String>|Object} local A list of font names to search. If an
- *   Object is passed, then local.alias is the name of an other substition font
- *   and local.append is a String to append to the list of fonts in the alias.
- *   For example if local.alias is "Foo" and local.append is "Bold" then the
- *   list of fonts will be "FooSubst1 Bold", "FooSubst2 Bold", etc.
- * @returns an String with the local fonts.
+ * Generate font description.
+ * @param {Object} param0, font substitution description.
+ * @param {Array<String>} src, contains src values (local(...) or url(...)).
+ * @param {String} localFontPath, path to local fonts.
+ * @param {boolean} useFallback, whether to use fallback font.
+ * @param {boolean} usePath, whether to use path to font.
+ * @param {String} append, style (Bold, Italic, ...) to append to font name.
+ * @return {Object} { style, ultimate }.
  */
-function makeLocal(prepend, appendToPrepended, local) {
-  let append = "";
-  if (!Array.isArray(local)) {
-    // We are getting our list of fonts in the alias and we'll append Bold,
-    // Italic or both.
-    append = ` ${local.append}`;
-    local = substitutionMap.get(local.alias).local;
+function generateFont(
+  { alias, local, path, fallback, style, ultimate },
+  src,
+  localFontPath,
+  useFallback = true,
+  usePath = true,
+  append = ""
+) {
+  const result = {
+    style: null,
+    ultimate: null,
+  };
+  if (local) {
+    const extra = append ? ` ${append}` : "";
+    for (const name of local) {
+      src.push(`local(${name}${extra})`);
+    }
   }
-  if (appendToPrepended) {
-    appendToPrepended = ` ${appendToPrepended}`;
+  if (alias) {
+    const substitution = substitutionMap.get(alias);
+    const aliasAppend = append || getStyleToAppend(style);
+    Object.assign(
+      result,
+      generateFont(
+        substitution,
+        src,
+        localFontPath,
+        /* useFallback = */ useFallback && !fallback,
+        /* usePath = */ usePath && !path,
+        aliasAppend
+      )
+    );
   }
-
-  let prependedPaths = "";
-  if (prepend) {
-    prependedPaths =
-      prepend.map(name => `local(${name}${appendToPrepended})`).join(",") + ",";
+  if (style) {
+    result.style = style;
+  }
+  if (ultimate) {
+    result.ultimate = ultimate;
+  }
+  if (useFallback && fallback) {
+    const fallbackInfo = substitutionMap.get(fallback);
+    const { ultimate: fallbackUltimate } = generateFont(
+      fallbackInfo,
+      src,
+      localFontPath,
+      useFallback,
+      /* usePath = */ usePath && !path,
+      append
+    );
+    if (!result.ultimate) {
+      result.ultimate = fallbackUltimate;
+    }
```

Comment:
Now that we're able to use modern JS also in the worker-thread, how about the following?
```suggestion
    result.ultimate ||= fallbackUltimate;
```